src/liballoc/allocator.rs

```diff
@@ -464,6 +464,27 @@ impl fmt::Display for CannotReallocInPlace {
 ///  * if a layout `k` fits a memory block (denoted by `ptr`)
 ///    currently allocated via an allocator `a`, then it is legal to
 ///    use that layout to deallocate it, i.e. `a.dealloc(ptr, k);`.
+///
+/// # Unsafety
+///
+/// The `Alloc` trait is an `unsafe` trait for a number of reasons, and
+/// implementors must ensure that they adhere to these contracts:
+///
+/// * Pointers returned from allocation functions must point to valid memory and
+///   retain their validity until at least the instance of `Alloc` is dropped
+///   itself.
+///
+/// * Implementations of `Alloc` are not currently memory safe if they unwind.
```

Comment:
Note that if I am wrong, and that this somehow is relevant to *all* implementations of `Alloc`, including ones that cannot be installed as global allocators, then I think more of the documentation for this trait needs to be revised.

E.g. text like this on `fn alloc` implies that an implementor may panic (even if it is not recommended): 
> Implementations are encouraged to return `Err` on memory exhaustion rather than panicking or aborting, but this is not a strict requirement.